packages/jsii-kernel/lib/kernel.ts

```diff
@@ -717,15 +717,15 @@ export class Kernel {
             const param = params[i];
             const arg = args[i];
 
-            if (param.variadic) {
+            if (spec.isVariadic(param)) {
                 if (params.length <= i) { return; } // No vararg was provided
                 for (let j = i ; j < params.length ; j++) {
-                    if (params[j] === undefined) {
+                    if (!param.type.nullable && params[j] === undefined) {
                         // tslint:disable-next-line:max-line-length
                         throw new Error(`Unexpected 'undefined' value at index ${j - i} of variadic argument '${param.name}' of type '${this._formatTypeRef(param.type)}'`);
                     }
                 }
-            } else if (!param.type.optional && arg === undefined) {
+            } else if (!param.type.nullable && arg === undefined) {
```

Comment:
This should be `param.optional` I think. This code is doing argument matching, not type matching.